zero_bin/rpc/src/jumpdest.rs

```diff
@@ -0,0 +1,471 @@
+        trace!("{:#?}\n", entry);
+
+        match op {
+            "CALL" | "CALLCODE" | "DELEGATECALL" | "STATICCALL" => {
+                ensure!(entry.stack.as_ref().is_some(), "No evm stack found.");
+                // We reverse the stack, so the order matches our assembly code.
+                let evm_stack: Vec<_> = entry.stack.as_ref().unwrap().iter().rev().collect();
+                let operands = 2; // actually 6 or 7.
+                ensure!(
+                    evm_stack.len() >= operands,
+                    "Opcode {op} expected {operands} operands at the EVM stack, but only {} were found.",
+                    evm_stack.len()
+                );
+                // This is the same stack index (i.e. 2nd) for all four opcodes. See https://ethervm.io/#F1
+                let [_gas, address, ..] = evm_stack[..] else {
+                    unreachable!()
+                };
+
+                let callee_address = {
+                    // Clear the upper half of the operand.
+                    let callee_raw = *address;
+                    // let (callee_raw, _overflow) = callee_raw.overflowing_shl(128);
+                    // let (callee_raw, _overflow) = callee_raw.overflowing_shr(128);
+
+                    ensure!(callee_raw <= U256::from(U160::MAX));
+                    let lower_20_bytes = U160::from(callee_raw);
+                    Address::from(lower_20_bytes)
+                };
+
+                if precompiles().contains(&callee_address) {
+                    trace!("Called precompile at address {}.", &callee_address);
+                } else if callee_addr_to_code_hash.contains_key(&callee_address) {
+                    let code_hash = callee_addr_to_code_hash[&callee_address];
+                    call_stack.push((code_hash, next_ctx_available));
+                } else {
+                    // This case happens if calling an EOA. This is described
+                    // under opcode `STOP`: https://www.evm.codes/#00?fork=cancun
+                    trace!(
+                        "Callee address {} has no associated `code_hash`.",
+                        &callee_address
+                    );
+                }
+                next_ctx_available += 1;
+                prev_jump = None;
+            }
+            "CREATE" => {
+                ensure!(entry.stack.as_ref().is_some(), "No evm stack found.");
+                // We reverse the stack, so the order matches our assembly code.
+                let evm_stack: Vec<_> = entry.stack.as_ref().unwrap().iter().rev().collect();
+                let operands = 3;
+                ensure!(
+                    evm_stack.len() >= operands,
+                    "Opcode {op} expected {operands} operands at  the EVM stack, but only {} were found.",
+                    evm_stack.len()
+                );
+                let [_value, _offset, _size, _salt, ..] = evm_stack[..] else {
+                    unreachable!()
+                };
```

Comment:
This got block 37 of the test chain to panic.

Reply:
Same for block 75

Reply:
This problem is now fixed, but the blocks still fail for other reasons.